### PR TITLE
chore: add flag for azcopy use sas token by default

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind test

**What this PR does / why we need it**:

test: add flag for azcopy use sas token by default

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/azurefile/azurefile_options.go [hunk 1]
```diff
@@ -81,6 +82,7 @@ func (o *DriverOptions) AddFlags() *flag.FlagSet {
 	fs.IntVar(&o.SasTokenExpirationMinutes, "sas-token-expiration-minutes", 1440, "sas token expiration minutes during volume cloning")
 	fs.StringVar(&o.KubeConfig, "kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
 	fs.StringVar(&o.Endpoint, "endpoint", "unix://tmp/csi.sock", "CSI endpoint")
+	fs.BoolVar(&o.AzcopyUseSasToken, "azcopy-use-sas-token", true, "Whether to use SAS token for azcopy")
```
Comment (andyzhangx) [inline pkg/azurefile/azurefile_options.go:85]: indicates whether SAS token should be used in azcopy based on volume clone and snapshot restore

Comment (umagnus) [reply pkg/azurefile/azurefile_options.go]: done

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**